packages/reactive-element/src/test/reactive-element_test.ts

```diff
@@ -2197,13 +2197,13 @@ suite('ReactiveElement', () => {
       promiseFulfilled = false;
 
       get updateComplete() {
-        return (async () => {
+        return (async (): Promise<boolean | void> => {
```

Comment:
Seems like you changed this to not resolve to `boolean`?

Reply:
Removed. The `updateComplete` promise previously resolved to `unknown`. This was changed and now reverted back.

It normally resolves to a boolean but the first one resolves to `undefined` and it it's maybe not worth changing that?

Reply:
Hmm, if we're going to resolve to a boolean, shouldn't the initial `updateComplete` correctly reflect the `isUpdatePending` state? If that's intended to be a useful thing, seems bad if you can't rely on that to know if a further update was enqueued on the first render, only on subsequent ones.

It sort of points out that the gambit of just making `enableProperties` be the initial `__updatePromise` resolver is maybe too simplistic, since we're not actually resolving after `performUpdate()` is done; it just sort of works because a user's `await el.updateComplete` will be the second thing queued behind the `await this.__updatePromise` in `__enqueueUpdate()`. We should really be waiting to resolve the `__updatePromise` after `performUpdate` when we know the `isUpdatePending` state.

It feels like we should be able to refactor a little to use the new `__pendingConnectionPromise` now to do the initial connection holdoff as well, maybe?

Let's make an issue and not block, but seems like we should fix that or remove the boolean, one or the other.